app/middleware/gp.js

```diff
@@ -0,0 +1,86 @@
+const util = require('util');
+const http = require('http');
+const gpDetailsParser = require('../utilities/gpDetailsParser');
+const gpOpeningTimesParser = require('../utilities/gpOpeningTimesParser');
+const daysOfTheWeek = require('../utilities/constants').daysOfTheWeek;
+
+function getDetails(req, res, next) {
+  http.get(req.urlForGp, (response) => {
+    let syndicationXml = '';
+    response.on('data', (chunk) => {
+      syndicationXml += chunk;
+    });
+
+    response.on('end', () => {
+      if (response.statusCode === 200) {
+        /* Disabled since assigning to res is recommended best practice my Express */
+        /* eslint-disable no-param-reassign */
+        req.gpDetails = gpDetailsParser(syndicationXml);
+        next();
+      } else if (response.statusCode === 404) {
+        const err = new Error('GP Not Found');
+        err.status = 404;
+        next(err);
+      } else {
+        next(`Error: ${response.statusCode}`);
+      }
+    });
+  }).on('error', (e) => {
+    console.log('Got an error: ', e);
+    next(e);
+  });
+}
+
+function getOpeningTimes(req, res, next) {
+  http.get(req.gpDetails.overviewLink, (response) => {
```

Comment:
I think this should be addressed as a general review of parameter validation and error handling in a separate piece of work to be done next. Will create an issue for this.